### ISSUE TITLE
sf project deploy quick command getting failed

### ISSUE THREAD

**Issue by shabu74** [forcedotcom/cli#2154]
I use the following command to perform a quick deploy:

sfdx force:source:deploy -q $deploy_id -w 0

It's working fine. But Salesforce is continuously warning me.

"We plan to deprecate this command in the future. Try using the "project deploy start" and "project deploy report" pair instead."

So I tried to use new project command:

sf project deploy quick --async --job-id $deploy_id

But I got the  following error:

{
 "code": 1,
 "context": "DeployMetadataQuick",
 "commandName": "DeployMetadataQuick",
 "message": "Job ID can't be used for quick deployment. Possible reasons include the deployment hasn't been validated or the validation expired because you ran it more than 10 days ago.",
 "name": "CannotQuickDeployError",
 "status": 1,
 "exitCode": 1,
 "warnings": []
}

Please note that deploy_id id the deployment job id I got for the the successful checkonly-deployment I have done before minutes and that job is still valid and it is working fine for force deploy command.

**Comment by iowillhoit** [comment]
Hello @shabu74, I am not seeing this issue. What version of `sf` are you using (`sf version --verbose`)?

With `sf 1.80.5` I did the folllowing:
- Cloned [Dreamhouse](https://github.com/dreamhouseapp/dreamhouse-lwc) and followed the setup steps
- Ran `sf project deploy validate --source-dir force-app`
- Then ran `sf project deploy quick --job-id THE_PROVIDED_ID`
- The deploy was successful
- I also try these commands with the `--async` flag and did not have issues

A few things that might be causing issues
- If you are running your `validate` command with `--async`, it might not have finished yet. 
- I noticed if you run the `sf project deploy quick` command more than once, the second time it "fails" because it has already deployed. The error message is not very clear on this though

If you are still seeing issues, please provide a repo (fork of Dreamhouse is fine) with the exact setup and commands to run to reproduce your issue.

**Comment by shabu74** [comment]
Hi mate, many thanks for your reply.

I missed one thing in my post. I did not use sf project deploy validate. I have used sf project deploy start --dry-run with async option. The reason is that I need to do destructive deployment too. sf project deploy validate does not have the option to specify destructive manifests.

But that deployment job id (getting using sf project deploy start command) works fine with sf force command for quick deployment, even with async option. That is why I am getting confused. 

However, you have pointed one issue "If you are running your validate command with --async, it might not have finished yet." 
So, let me try f project deploy start --dry-run with async, but with -w 1. I will come back to you after that.
